eox_tagging/models.py

```diff
@@ -97,14 +103,19 @@ def __get_object_for_this_type(self, object_type, object_id):
             }
 
         if object_type == COURSE_ENROLLMENT_MODEL_NAME:
+            course_id = object_id.get("course_id")
             object_id = {
-                "course_id": CourseKey.from_string(object_id.get("course_id")),
-                "user__username": object_id.get("username"),
+                name: value for name, value in
+                (
+                    ("course_id", CourseKey.from_string(course_id) if course_id else None),
+                    ("user__username", object_id.get("username")),
+                )
+                if value is not None
```

Comment:
This block is a bit hard to read. I think we can do without the dict comprehension, given that you extracting from tuples instead of an iterator you can use something like this instead:

```suggestion
            if value:
                object_id = {
                    "course_id": CourseKey.from_string(course_id) if course_id else None,
                    "user__username": object_id.get("username")
                }
```

Reply:
Yes that could be better